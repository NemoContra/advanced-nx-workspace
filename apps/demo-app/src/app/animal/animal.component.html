<p>A cat or a dog</p>
<flight-workspace-shared-util-random-animal-image></flight-workspace-shared-util-random-animal-image>
