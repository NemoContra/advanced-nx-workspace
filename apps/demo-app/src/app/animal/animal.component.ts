import { Component, OnInit } from '@angular/core';

@Component({
  selector: 'flight-workspace-animal',
  templateUrl: './animal.component.html',
  styleUrls: ['./animal.component.css'],
})
export class AnimalComponent {}
