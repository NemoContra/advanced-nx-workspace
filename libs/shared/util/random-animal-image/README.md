# shared-util-random-animal-image

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test shared-util-random-animal-image` to execute the unit tests.
