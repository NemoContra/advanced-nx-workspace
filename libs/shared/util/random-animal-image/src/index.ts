export * from './lib/shared-util-random-animal-image.module';
