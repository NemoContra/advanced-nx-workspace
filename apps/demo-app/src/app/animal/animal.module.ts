import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { AnimalComponent } from './animal.component';
import { SharedUtilRandomAnimalImageModule } from '@flight-workspace/shared/util/random-animal-image';
import { RouterModule } from '@angular/router';

@NgModule({
  declarations: [AnimalComponent],
  imports: [
    CommonModule,
    SharedUtilRandomAnimalImageModule,
    RouterModule.forChild([
      {
        path: '',
        component: AnimalComponent,
      },
    ]),
  ],
  exports: [AnimalComponent],
})
export class AnimalModule {}
