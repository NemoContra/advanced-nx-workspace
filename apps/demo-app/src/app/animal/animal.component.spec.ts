import { createRoutingFactory } from '@ngneat/spectator/jest';
import { AnimalComponent } from './animal.component';
import { MockModule } from 'ng-mocks';
import { SharedUtilRandomAnimalImageModule } from '@flight-workspace/shared/util/random-animal-image';

describe('AnimalComponent', () => {
  const createComponent = createRoutingFactory({
    component: AnimalComponent,
    routes: [
      {
        path: '',
        component: AnimalComponent,
      },
    ],
    imports: [MockModule(SharedUtilRandomAnimalImageModule)],
  });

  it('should render', () => {
    const spectator = createComponent();
    expect(spectator.fixture).toMatchSnapshot();
  });
});
