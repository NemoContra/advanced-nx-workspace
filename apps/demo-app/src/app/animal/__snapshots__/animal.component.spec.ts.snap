// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimalComponent should render 1`] = `
<flight-workspace-animal
  __ngContext__={[Function LRootView]}
>
  <p>
    A cat or a dog
  </p><flight-workspace-shared-util-random-animal-image />
</flight-workspace-animal>
`;
